src/path.rs

```diff
@@ -217,6 +217,7 @@ impl ClioPath {
     /// assert_eq!(ClioPath::new("/tmp/log.txt.gz")?, p);
     /// # Ok::<(), clio::Error>(())
     /// ```
+    #[rustversion::since(1.70)]
```

Comment:
I'm not sure it is worth adding the dependency and second implementation just to avoid one allocation, but I am open to being persuaded otherwise.